src/commands/transaction/secretlock.ts

```diff
@@ -0,0 +1,171 @@
+import chalk from 'chalk';
+import { command, metadata, option } from 'clime';
+import {
+    Account,
+    Address,
+    BlockHttp,
+    Deadline,
+    Mosaic,
+    MosaicId,
+    SecretLockTransaction,
+    TransactionHttp,
+    UInt64,
+} from 'nem2-sdk';
+import { AnnounceTransactionsCommand, AnnounceTransactionsOptions } from '../../announce.transactions.command';
+import { OptionsResolver } from '../../options-resolver';
+import { SecretLockService } from '../../service/secretlock.service';
+import { AddressValidator } from '../../validators/address.validator';
+import { HashAlgorithmValidator } from '../../validators/hashAlgorithm.Validator';
+import { MosaicIdValidator } from '../../validators/mosaicId.validator';
+import { NetworkValidator } from '../../validators/network.validator';
+
+export class CommandOptions extends AnnounceTransactionsOptions {
+    @option({
+        description: 'Locked mosaic ID.',
+        flag: 'm',
+        validator: new MosaicIdValidator(),
+    })
+    mosaicId: string;
+
+    @option({
+        description: '(Optional) Amount to lock.',
+        flag: 'a',
+        default: 10,
+    })
+    amount: number;
+
+    @option({
+        description: 'Number of blocks for which a lock should be valid. ' +
+            'Duration is allowed to lie up to 30 days. If reached, the mosaics will be returned to the initiator.',
+        flag: 'd',
+    })
+    duration: number;
```

Comment:
Change type for "string" and add the "NumericString" validator to allow more representation.